src/libs/ReportUtils.ts

```diff
@@ -559,42 +501,34 @@ function findLastAccessedReport(reports, ignoreDomainRooms, policies, isFirstTim
             return sortedReports[0];
         }
 
-        return adminReport || _.find(sortedReports, (report) => !isConciergeChatReport(report));
+        return adminReport ?? sortedReports.find((report) => !isConciergeChatReport(report));
     }
 
     if (ignoreDomainRooms) {
         // We allow public announce rooms, admins, and announce rooms through since we bypass the default rooms beta for them.
         // Check where ReportUtils.findLastAccessedReport is called in MainDrawerNavigator.js for more context.
         // Domain rooms are now the only type of default room that are on the defaultRooms beta.
-        sortedReports = _.filter(
-            sortedReports,
-            (report) => !isDomainRoom(report) || getPolicyType(report, policies) === CONST.POLICY.TYPE.FREE || hasExpensifyGuidesEmails(lodashGet(report, ['participantAccountIDs'], [])),
+        sortedReports = sortedReports.filter(
+            (report) => !isDomainRoom(report) || getPolicyType(report, policies) === CONST.POLICY.TYPE.FREE || hasExpensifyGuidesEmails(report?.participantAccountIDs ?? []),
         );
     }
 
-    return adminReport || _.last(sortedReports);
+    return adminReport ?? sortedReports[sortedReports.length - 1];
 }
 
 /**
  * Whether the provided report is an archived room
- * @param {Object} report
- * @param {Number} report.stateNum
- * @param {Number} report.statusNum
- * @returns {Boolean}
  */
-function isArchivedRoom(report) {
-    return report && report.statusNum === CONST.REPORT.STATUS.CLOSED && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED;
+function isArchivedRoom(report: OnyxEntry<Report> | undefined): boolean {
```

Comment:
Is undefined needed here? Can we get rid of it?

Reply:
I think we should leave it 
this function its used for example here
<img width="680" alt="image" src="https://github.com/Expensify/App/assets/25230417/91a5a894-53d4-4482-8545-18a8842d14bb">
where `report.parentReportID` is possibly undefined so `getReport` function will return undefined